### ISSUE TITLE
Build on Linux error (mono/C# incompatible?)

### ISSUE THREAD

**Issue by xenogenesi** [FlaxEngine/FlaxEngine#653]
I saw the issue https://github.com/FlaxEngine/FlaxEngine/issues/61, I get the same error message, but the problem seems to me to be a different one.

The system is debian/sid (from experience I expect it to behave mostly like ubuntu, but I'm not a mono/C# expert), I installed clang++-9.

From what I have noticed the comparison of the result of the `Which()! = null` function does not work, see log below, I have added a print in `LinuxPlatform()`, note that all if else are executed and at the end Compiler is null:

```
[00:00:076] Loaded project Flax, version 1.2.6226
[00:00:339] Missing Android SDK. Cannot build for Android platform.
[00:00:342] which clang++-10 exit code: 1, result: 
[00:00:344] which clang++-9 exit code: 0, result: /usr/bin/clang++-9
[00:00:345] which clang++-8 exit code: 1, result: 
[00:00:347] which clang++-7 exit code: 1, result: 
[00:00:349] which clang++-6 exit code: 1, result: 
[00:00:350] which clang++ exit code: 0, result: /usr/lib/llvm-9/bin/clang++
[00:00:351] Compiler == null
[00:00:359] Building targets...
```

if I use string.Empty instead of null it works and I proceed to the next error:

```
Building targets...
Exception: Missing Clang (x86_64-unknown-linux-gnu/bin/clang++)
```

I probably have the same problem with `systemCompiler` in` UnixToolchain`, I'm trying to figure out where it comes from.

One note, although `llvm-ar` is installed it is not in the `PATH` by default, it is in`/usr/lib/llvm-9/bin/llvm-ar`, in the output above `clang++` is found in that path because I had added the directory in `PATH`.

**Comment by stefnotch** [comment]
Glad to hear that you solved this issue. So, if I understood correctly, the entire error was caused by a missing Vulkan SDK?
Or is that a separate issue?

**Comment by xenogenesi** [comment]
@stefnotch no, once the compilation worked, the editor seemed to hang at runtime (no GUI appears), only this issue was caused by the lack of the Vulkan SDK (I didn't read anywhere that it was essential, maybe it should be added to the documentation for the Linux build? Maybe with a link where to download them, luckily I found them in a closed issue).

What appeared to be some strange behavior of mono/C#, was due to the fact that the `which` command in Debian/sid is deprecated (zsh probably uses a builtin). The Which() function appeared to work in the logs, but as there was output in stderr, it returned null. For this reason, no toolchain command was found.

There are a couple of minor changes I had to make to be able to compile.

The `v` variable in AndroidNdk.cs:

```diff
- if (Version.TryParse (ver, out var v))
+ if (Version.TryParse (ver, out var vc))
                         {
- Version = v;
+ Version = vc;
```

and getpagesize() in ThirdParty/tracy/libbacktrace/mmapio.cpp:

```diff
-extern int getpagesize (void);
+ // extern int getpagesize (void);
+ // error: 'getpagesize' is missing exception specification 'throw ()'
+ # define getpagesize () sysconf (_SC_PAGESIZE)
```

Another note: in the current documentation it says that there are vscode tasks to perform the build, in reality I do not see tasks in the .vscode/tasks.json file, to compile the editor I am using the command that in .github/workflows/build_linux.xml.

```
./Development/Scripts/Linux/CallBuildTool.sh -build -log -arch = x64 -platform = Linux -configuration = Development -buildtargets = FlaxEditor
```
